File: Maximum Intersecting Lines[ 10 April 2023]/Approach2(Map).java
//	>> JAVA CODE <<

public int maxIntersections(int[][] lines, int N) {
       
       TreeMap<Integer, Integer> map = new TreeMap<>();
       
        for(int[] line : lines){
            int start = line[0], end = line[1] + 1;
            map.putIfAbsent(start, 0);
            map.putIfAbsent(end, 0);
            
            map.put(start, map.get(start) + 1);
            map.put(end, map.get(end) - 1);
        }
        
        int res = 0, cnt = 0;
        for(Integer value:map.values()){
            cnt += value;
            res = Math.max(res, cnt);
        }
        
        return res;
       
    }

File: Maximum Intersecting Lines[ 10 April 2023]/Approach2(Map).py
#  >> PYTHON CODE <<

def maxIntersections(self, lines, N):
        lineMap = dict()
        for line in lines:
            start = line[0]
            end = line[1] + 1
        
            if start in lineMap:
                lineMap[start]+=1
            else: 
                lineMap[start]=1
            if end in lineMap:
                lineMap[end]-=1
            else:
                lineMap[end]=-1
        
        res = 0
        cnt = 0
        for entry in sorted(lineMap.keys()):
            cnt += lineMap[entry]
            res = max(res, cnt)
        
        return res